[Bug] [SQL] after sql replaced, comment remained part, cause error, only doris driver?

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

**this is doris data source, doris driver** sql comment replaced wrong
---------------------------------------------
```
[INFO] 2025-02-19 12:16:39.612 +0800 - *********************************  Execute task instance  *************************************
[INFO] 2025-02-19 12:16:39.613 +0800 - ***********************************************************************************************
[INFO] 2025-02-19 12:16:39.613 +0800 - Full sql parameters: SqlParameters{type='DORIS', datasource=36, sql='--注释测试
/*注释测试*/
select 'a' as s,--注释测试
/*注释测试*/
 1 as n
 --注释测试
/*注释测试*/', sqlType=0, sendEmail=null, displayRows=10, limit=0, udfs='null', showType='null', connParams='null', groupId='0', title='null', preStatements=[], postStatements=[]}
[INFO] 2025-02-19 12:16:39.613 +0800 - sql type : DORIS, datasource : 36, sql : --注释测试
/*注释测试*/
select 'a' as s,--注释测试
/*注释测试*/
 1 as n
 --注释测试
/*注释测试*/ , localParams : [],udfs : null,showType : null,connParams : null,varPool : [] ,query max result limit  0
[INFO] 2025-02-19 12:16:39.615 +0800 - after replace sql , preparing : 测试*/
select 'a' as s,
测试*/
 1 as n
 
测试*/
[INFO] 2025-02-19 12:16:39.616 +0800 - Sql Params are replaced sql , parameters:
[INFO] 2025-02-19 12:16:39.616 +0800 - can't find udf function resource
[WARN] 2025-02-19 12:16:39.617 +0800 - Connect strings must start with jdbc:snowflake://
[INFO] 2025-02-19 12:16:39.662 +0800 - prepare statement replace sql : 测试*/
select 'a' as s,
测试*/
 1 as n
 
测试*/, sql parameters : {}
[INFO] 2025-02-19 12:16:39.662 +0800 - main statement execute query, for sql: 测试*/
select 'a' as s,
测试*/
 1 as n
 
测试*/
[ERROR] 2025-02-19 12:16:39.669 +0800 - execute sql error: errCode = 2, detailMessage = 
mismatched input '测试' expecting {<EOF>, ';', '(', 'ADMIN', 'ALTER', 'ANALYZE', 'BACKUP', 'BEGIN', 'BUILD', 'CALL', 'CANCEL', 'CLEAN', 'COLUMNS', 'COMMIT', 'COPY', 'CREATE', 'DELETE', 'DESC', 'DESCRIBE', 'DROP', 'EXPLAIN', 'EXPORT', 'GRANT', 'HELP', 'INSERT', 'INSTALL', 'KILL', 'LOAD', 'LOCK', 'ORDER', 'PAUSE', 'PLAN', 'PRECEDING', 'RECOVER', 'REFRESH', 'REPLACE', 'RESTORE', 'RESUME', 'REVOKE', 'ROLLBACK', 'SELECT', 'SET', 'SHOW', 'START', 'STOP', 'SWITCH', 'SYNC', 'TRUNCATE', 'UNINSTALL', 'UNLOCK', 'UNSET', 'UPDATE', 'USE', 'VALUES', 'WARM', 'WHERE', 'WITH'}(line 1, pos 0)

[ERROR] 2025-02-19 12:16:39.670 +0800 - sql task error
```

**this is doris data source, mysql driver**
---------------------------------------------
```
[INFO] 2025-02-19 12:23:52.946 +0800 - *********************************  Execute task instance  *************************************
[INFO] 2025-02-19 12:23:52.947 +0800 - ***********************************************************************************************
[INFO] 2025-02-19 12:23:52.947 +0800 - Full sql parameters: SqlParameters{type='MYSQL', datasource=7, sql='--注释测试
/*注释测试*/
select 'a' as s,--注释测试
/*注释测试*/
 1 as n
 --注释测试
/*注释测试*/', sqlType=0, sendEmail=null, displayRows=10, limit=0, udfs='null', showType='null', connParams='null', groupId='0', title='null', preStatements=[], postStatements=[]}
[INFO] 2025-02-19 12:23:52.950 +0800 - sql type : MYSQL, datasource : 7, sql : --注释测试
/*注释测试*/
select 'a' as s,--注释测试
/*注释测试*/
 1 as n
 --注释测试
/*注释测试*/ , localParams : [],udfs : null,showType : null,connParams : null,varPool : null ,query max result limit  0
[INFO] 2025-02-19 12:23:52.951 +0800 - after replace sql , preparing : --注释测试

select 'a' as s,--注释测试

 1 as n
 --注释测试
[INFO] 2025-02-19 12:23:52.951 +0800 - Sql Params are replaced sql , parameters:
[INFO] 2025-02-19 12:23:52.951 +0800 - can't find udf function resource
[WARN] 2025-02-19 12:23:52.952 +0800 - Connect strings must start with jdbc:snowflake://
[INFO] 2025-02-19 12:23:52.990 +0800 - prepare statement replace sql : --注释测试

select 'a' as s,--注释测试

 1 as n
 --注释测试, sql parameters : {}
[INFO] 2025-02-19 12:23:52.990 +0800 - main statement execute query, for sql: --注释测试

select 'a' as s,--注释测试

 1 as n
 --注释测试
[INFO] 2025-02-19 12:23:53.054 +0800 - display sql result 1 rows as follows:
[INFO] 2025-02-19 12:23:53.055 +0800 - row 1 : {"s":"a","n":1}
```

### What you expected to happen

comment replaced correct
dolphin scheduler 3.2.2

### How to reproduce

sql type job, doris driver data source
```
--注释测试
/*注释测试*/
select 'a' as s,--注释测试
/*注释测试*/
 1 as n
 --注释测试
/*注释测试*/
```

### Anything else

_No response_

### Version

3.2.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-doris/src/main/java/org/apache/dolphinscheduler/plugin/doris/param/DorisDataSourceProcessor.java
+++ b/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-doris/src/main/java/org/apache/dolphinscheduler/plugin/doris/param/DorisDataSourceProcessor.java
@@ -37,6 +37,7 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
+import com.alibaba.druid.sql.parser.SQLParserUtils;
 import com.google.auto.service.AutoService;
 
 @AutoService(DataSourceProcessor.class)
@@ -137,6 +138,14 @@ public class DorisDataSourceProcessor extends AbstractDataSourceProcessor {
     @Override
     public DataSourceProcessor create() {
         return new DorisDataSourceProcessor();
+    }
+
+    @Override
+    public List<String> splitAndRemoveComment(String sql) {
+        // Because doris is highly compatible with mysql syntax,
+        // a mysql-type implementation can be used for comments and statement split.
+        String cleanSQL = SQLParserUtils.removeComment(sql, com.alibaba.druid.DbType.mysql);
+        return SQLParserUtils.split(cleanSQL, com.alibaba.druid.DbType.mysql);
     }
 
     private String transformOther(Map<String, String> paramMap) {

--- a/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-doris/src/test/java/org/apache/dolphinscheduler/plugin/doris/param/DorisDataSourceProcessorTest.java
+++ b/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-doris/src/test/java/org/apache/dolphinscheduler/plugin/doris/param/DorisDataSourceProcessorTest.java
@@ -137,4 +137,15 @@ public class DorisDataSourceProcessorTest {
         Assertions.assertEquals("set enable_unique_key_partial_update = false", sqls.get(2));
     }
 
+    @Test
+    public void testSplitAndRemoveCommentWithComments() {
+        String sql = "-- This is a comment\r\n" +
+                "select *, udfTest(p1) from table1 -- This is a comment\r\n;" +
+                "/*Annotation test*/select * from table2/*Annotation test*/";
+        List<String> result = dorisDatasourceProcessor.splitAndRemoveComment(sql);
+        Assertions.assertEquals(2, result.size());
+        Assertions.assertEquals("select *, udfTest(p1) from table1", result.get(0));
+        Assertions.assertEquals("select * from table2", result.get(1));
+    }
+
 }

--- a/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-doris/src/test/java/org/apache/dolphinscheduler/plugin/doris/param/DorisDataSourceProcessorTest.java
+++ b/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-doris/src/test/java/org/apache/dolphinscheduler/plugin/doris/param/DorisDataSourceProcessorTest.java
@@ -138,7 +138,7 @@ public class DorisDataSourceProcessorTest {
     }
 
     @Test
-    public void testSplitAndRemoveCommentWithComments() {
+    void testSplitAndRemoveCommentWithComments() {
         String sql = "-- This is a comment\r\n" +
                 "select *, udfTest(p1) from table1 -- This is a comment\r\n;" +
                 "/*Annotation test*/select * from table2/*Annotation test*/";

